app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt

```diff
@@ -90,322 +90,403 @@ class TopicPracticeFragmentTest {
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
-          1, R.id.subtopic_check_box
+          1,
+          R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.topic_practice_skill_list)).perform(
         scrollToPosition<RecyclerView.ViewHolder>(
-          10
+          5
         )
       )
-      onView(withId(R.id.topic_practice_start_button)).check(matches(isClickable()))
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          5,
+          R.id.topic_practice_start_button
+        )
+      ).check(matches(isClickable()))
     }
   }
 
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_deselectSubtopics_isSuccessful() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          R.id.topic_practice_skill_list,
+          1,
+          R.id.subtopic_check_box
+        )
+      ).perform(click())
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
           1, R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
           1, R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).check(matches(not(isChecked())))
     }
   }
 
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_deselectsubtopics_startButtonIsInactive() { // ktlint-disable max-line-length
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
           1, R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
       onView(
         atPositionOnView(
           R.id.topic_practice_skill_list,
           1, R.id.subtopic_check_box
         )
-      ).perform(
-        click()
-      )
+      ).perform(click())
       onView(withId(R.id.topic_practice_skill_list)).perform(
         scrollToPosition<RecyclerView.ViewHolder>(
-          10
```

Comment:
Why did the index change? Is this a screen size thing? Will this be flaky across different screen sizes?

Reply:
10 is wrong previously because it's not the real index of the last item in the recyclerview.